substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/Safepoint.java

```diff
@@ -657,15 +662,17 @@ private static boolean isMyself(IsolateThread thread) {
         }
 
         /** Send each of the threads (except myself) a request to come to a safepoint. */
-        private static void requestSafepoints(String reason) {
+        private static int requestSafepoints(String reason) {
             VMThreads.THREAD_MUTEX.assertIsOwner("Must hold mutex while requesting a safepoint.");
             final Log trace = Log.noopLog().string("[Safepoint.Master.requestSafepoints:  reason: ").string(reason);
+            int numOfThreads = 0;
 
             // Walk the threads list and ask each thread (except myself) to come to a safepoint.
             for (IsolateThread vmThread = VMThreads.firstThread(); vmThread.isNonNull(); vmThread = VMThreads.nextThread(vmThread)) {
                 if (isMyself(vmThread)) {
                     continue;
                 }
+                numOfThreads++;
```

Comment:
Shouldn't this count also include the current thread? This code is always executed by the VM operation thread but the VM operation thread is a normal Java thread in Native Image. We support two different execution modes (see `VMOperationControl.useDedicatedVMOperationThread()`):
a. Without a dedicated VM operation thread, any normal Java thread can be temporarily promoted to the VM operation thread.
b. If a dedicated VM operation thread should be used, then a dedicated Java thread is started early on and only that thread may execute VM operations.

Reply:
I believe it means how many threads were brought to the safepoint. In hotspot, it does not count vmThread.

Reply:
HotSpot reports the number of Java threads, so I think we should the same.
However, in our case that includes the VM operation thread because it is a Java thread as well. This is a major difference to HotSpot anyways and affects all JFR areas so I think it is best to consistently report the VM operation thread as a normal Java thread (e.g., the VM operation thread can allocate Java heap memory, so it will also show up in all allocation-related JFR events).

Reply:
Okay, fixed

Reply:
Thanks, this PR will also be merged to Graal master in the next couple days.